### PR TITLE
Icon a11y guide 2 UXPLATFORM-6244

### PR DESCRIPTION
**Pull request by xenoworf**
### Summary
This guide shows our consumers how to use terra-icons correctly so that more users can access their content.

### Deployment Link
https://terra-core-icon-a11y-gu-ghpi06.herokuapp.com/components/cerner-terra-core-docs/icon/accessibility-guide

### Testing
N/A - it's doc.

### Other tracking links
https://jira2.cerner.com/browse/UXPLATFORM-6244

### CODE REVIEW

File: packages/terra-core-docs/src/terra-dev-site/doc/icon/AccessibilityGuide.2.doc.mdx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    <div role="group" aria-labelledby="controlsLabel">
+        <span id="controlsLabel">Playback controls</span><br/>
+        <button><IconMediaRewind a11yLabel="Rewind" /></button>
+        <button><IconMediaPlay a11yLabel="Play" /></button>
+        <button><IconMediaFastForward a11yLabel="Fast Forward" /></button>
+    </div>
+
+</div>
+
+The screen reader user will hear something like: <samp> Group: Playback Controls. Link image Rewind. Link image Play. Link Fast image Forward.</samp>
```
Comment (jeremyfuksa) [inline packages/terra-core-docs/src/terra-dev-site/doc/icon/AccessibilityGuide.2.doc.mdx:116]: "Link image Fast Forward."

File: packages/terra-core-docs/src/terra-dev-site/doc/icon/AccessibilityGuide.2.doc.mdx [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+- None identified
+- [Report a problem with this component on **GitHub**](https://github.com/cerner/terra-core/issues/new/choose)
+
+_For more information about Accessibility Support with Terra — go to [Component Standards: Accessibility (A11y)](https://engineering.cerner.com/terra-ui/about/terra-ui/component-standards#accessibility-a11y)._
+
+## Linked References:
+
+1. Eggert, Eric; Abou-Zahra, Shadi; et al. (27 July 2019). ["Web Accessibility Tutorials: Image Concepts"](https://www.w3.org/WAI/tutorials/images/). World Wide Web Consortium. Last updated 27 July 2019. Retrieved 2 March 2022.
+
+2. Deque staff (21 March 2019). ["Summary and Checklist: Images, Canvas, SVG, and Other Non-Text Content"](https://dequeuniversity.com/assets/pdf/module-nontext/module-nontext-checklist.pdf). Deque University. Publised Version 2019.03.21. Retrieved 2 March 2022.
```
Comment (jeremyfuksa) [inline packages/terra-core-docs/src/terra-dev-site/doc/icon/AccessibilityGuide.2.doc.mdx:239]: "Published"

**Review by jeremyfuksa — APPROVED** [review]
Fixed two small typos. Approved with latest commit.

File: packages/terra-core-docs/CHANGELOG.md [hunk 3]
```diff
@@ -4,10 +4,13 @@
 
 * Changed
   * linter fixes.
+  * Terra Icon accessibility guide.
```
Comment (sdadn) [inline packages/terra-core-docs/CHANGELOG.md:7]: is this redundant?

File: packages/terra-core-docs/src/terra-dev-site/doc/icon/AccessibilityGuide.2.doc.mdx [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+    <VisuallyHiddenText text="Four out of five stars" />
+    <button><Star a11yLabel="Rate one star" /></button>
+    <button><Star a11yLabel="Rate two stars" /></button>
+    <button><Star a11yLabel="Rate three stars" /></button>
+    <button><Star a11yLabel="Rate four stars" /></button>
+    <button><EmptyStar a11yLabel="Rate five stars" /></button>
+
+</div>
+
+This example shows important differences in how to handle an interactive group of icons: That the user can click on any of the stars (links) to change the rating. You must convey what will happen when the user clicks on each icon. The meaning of the rating is conveyed by the `<VisuallyHiddenText />` given first. Each star icon <span role="presentation">(<FilledStar/>, and <EmptyStar/>)</span> describes it's own action, e.g. rating an item one star.
```
Comment (benbcai) [inline packages/terra-core-docs/src/terra-dev-site/doc/icon/AccessibilityGuide.2.doc.mdx:170]:
```suggestion
This example shows important differences in how to handle an interactive group of icons: That the user can click on any of the stars (links) to change the rating. You must convey what will happen when the user clicks on each icon. The meaning of the rating is conveyed by the `<VisuallyHiddenText />` given first. Each star icon <span role="presentation">(<FilledStar/> and <EmptyStar/>)</span> describes it's own action, e.g. rating an item one star.
```

**Review by benbcai — APPROVED** [review]

File: packages/terra-core-docs/CHANGELOG.md (hunk 3 above)
Comment (xenoworf) [reply packages/terra-core-docs/CHANGELOG.md]: Removed in b9e422e34d1181a337047253bf1dda4e510c656a

**Pull request merged**